torchrl/envs/libs/pettingzoo.py

```diff
@@ -0,0 +1,929 @@
+
+        >>> print(env.rand_action(env.reset()))
+        TensorDict(
+            fields={
+                agent: TensorDict(
+                    fields={
+                        action: Tensor(shape=torch.Size([3, 9]), device=cpu, dtype=torch.int64, is_shared=False),
+                        action_mask: Tensor(shape=torch.Size([3, 9]), device=cpu, dtype=torch.bool, is_shared=False),
+                        done: Tensor(shape=torch.Size([3, 1]), device=cpu, dtype=torch.bool, is_shared=False),
+                        observation: Tensor(shape=torch.Size([3, 3, 3, 2]), device=cpu, dtype=torch.int8, is_shared=False)},
+                    batch_size=torch.Size([3]))},
+                adversary: TensorDict(
+                    fields={
+                        action: Tensor(shape=torch.Size([9]), device=cpu, dtype=torch.int64, is_shared=False),
+                        action_mask: Tensor(shape=torch.Size([9]), device=cpu, dtype=torch.bool, is_shared=False),
+                        done: Tensor(shape=torch.Size([1]), device=cpu, dtype=torch.bool, is_shared=False),
+                        observation: Tensor(shape=torch.Size([3, 3, 2]), device=cpu, dtype=torch.int8, is_shared=False)},
+                    batch_size=torch.Size([]))},
+            batch_size=torch.Size([]))
+        >>> print(env.group_map)
+        {"agent": ["agent_0", "agent_1", "agent_2"], "adversary": ["adversary_0"]}
+
+    Otherwise, a group map can be specified or selected from some premade options.
+    See :class:`torchrl.envs.utils.MarlGroupMapType` for more info.
+    For example, you can provide ``MarlGroupMapType.ONE_GROUP_PER_AGENT``, telling that each agent should
+    have its own tensordict (similar to the pettingzoo parallel API).
+
+    Grouping is useful for leveraging vectorisation among agents whose data goes through the same
+    neural network.
+
+    Args:
+        env (``pettingzoo.utils.env.ParallelEnv`` or ``pettingzoo.utils.env.AECEnv``): the pettingzoo environment to wrap.
+        return_state (bool, optional): whether to return the global state from pettingzoo
+            (not available in all environments). Default False.
+        group_map (MarlGroupMapType or Dict[str, List[str]]], optional): how to group agents in tensordicts for
+            input/output. By default, agents will be grouped by their name. Otherwise, a group map can be specified
+            or selected from some premade options. See :class:`torchrl.envs.utils.MarlGroupMapType` for more info.
+        use_action_mask (bool, optional): whether the environment should ouptut an "action_mask". This is compulsory in
+            wrapped ``pettingzoo.AECEnv`` to mask out unavailable actions and non-acting agents and should be also used
+            for ``pettingzoo.ParallelEnv`` when the number of agents can vary. Default False.
+        seed (int, optional): the seed. Default None.
```

Comment:
```suggestion
        seed (int, optional): the seed. Defaults to ``None``.
```

Reply:
We should move away from seeding in constructors IMO (seeding should be done via set_seed whenever we can)

Reply:
but does it hurt to set it from the constructor? cause for envs like SMAC (you will see in the other PR) it can only be set at construction time